src/halmos/sevm.py

```diff
@@ -1348,7 +1356,35 @@ def arith(self, ex: Exec, op: int, w1: Word, w2: Word) -> Word:
             return w1 - w2
 
         if op == EVM.MUL:
-            return w1 * w2
+            if is_bv_value(w1) and is_bv_value(w2):
+                return w1 * w2
+
+            if is_bv_value(w1):
+                i1: int = w1.as_long()
+                if i1 == 0:
+                    return w1
+
+                if i1 == 1:
+                    return w2
+
+                if is_power_of_two(i1):
+                    return w2 << int(math.log(i1, 2))
```

Comment:
```suggestion
                    return w2 << i1.bit_length() - 1
```

quick check:

~/projects/halmos on code-fastpath wip *21 !2 ?50 ❯ python -m timeit --setup='import math' 'int(math.log(32, 2))'                                                                                                                                                                     
2000000 loops, best of 5: 121 nsec per loop

~/projects/halmos on code-fastpath wip *21 !2 ?50 ❯ python -m timeit 'int(32).bit_length() - 1'                                                                                                                                                                 
5000000 loops, best of 5: 66.1 nsec per loop

Reply:
done: https://github.com/a16z/halmos/pull/340/commits/2ee41207c8a111980670ff7f426fc20f1e57b89a